sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/query/CollectionQueryProvider.java

```diff
@@ -0,0 +1,141 @@
+    // TODO this was not existing before in DocumentDB
+    QueryOuterClass.QueryParameters parameters = getQueryParameters();
+
+    // TODO can we utilize the ifNotExist
+    return new QueryBuilder()
+        .create()
+        .table(namespace, collection)
+        .column(columns)
+        .parameters(parameters)
+        .build();
+  }
+
+  /**
+   * Provides query for deleting the collection.
+   *
+   * @param namespace Namespace of the collection.
+   * @param collection Collection name.
+   * @return Query
+   */
+  public QueryOuterClass.Query deleteCollectionQuery(String namespace, String collection) {
+    // parameters for the local quorum
+    // TODO this was not existing before in DocumentDB
+    QueryOuterClass.QueryParameters parameters = getQueryParameters();
+
+    // construct delete query
+    return new QueryBuilder().drop().table(namespace, collection).parameters(parameters).build();
+  }
+
+  /**
+   * Creates a set of queries to be performed for creating all needed indexes for a collection.
+   *
+   * @param namespace
+   * @param collection
+   * @return
+   */
+  public List<QueryOuterClass.Query> createCollectionIndexQueries(
+      String namespace, String collection) {
+    DocumentTableProperties tableProperties = documentProperties.tableProperties();
+    List<QueryOuterClass.Query> indexQueries = new ArrayList<>();
+
+    if (dataStoreProperties.saiEnabled()) {
+      indexQueries.add(
+          createIndexQuery(
+              namespace,
+              collection,
+              tableProperties.leafColumnName(),
+              STORAGE_ATTACHED_INDEX_CLASS));
+      indexQueries.add(
+          createIndexQuery(
+              namespace,
+              collection,
+              tableProperties.stringValueColumnName(),
+              STORAGE_ATTACHED_INDEX_CLASS));
+      indexQueries.add(
+          createIndexQuery(
+              namespace,
+              collection,
+              tableProperties.doubleValueColumnName(),
+              STORAGE_ATTACHED_INDEX_CLASS));
+      // SAI doesn't support booleans, so add a non-SAI index here.
+      indexQueries.add(
+          createIndexQuery(namespace, collection, tableProperties.booleanValueColumnName(), null));
+    } else {
+      indexQueries.add(
+          createIndexQuery(namespace, collection, tableProperties.leafColumnName(), null));
+      indexQueries.add(
+          createIndexQuery(namespace, collection, tableProperties.stringValueColumnName(), null));
+      indexQueries.add(
+          createIndexQuery(namespace, collection, tableProperties.doubleValueColumnName(), null));
+      indexQueries.add(
+          createIndexQuery(namespace, collection, tableProperties.booleanValueColumnName(), null));
+    }
+
+    return indexQueries;
+  }
+
+  private QueryOuterClass.Query createIndexQuery(
+      String keyspaceName, String tableName, String indexedColumn, String customIndexClass) {
+    // TODO Do I need a quorum here as well?
```

Comment:
I think it's OK not to wait for quorum on indexes - performance may suffer for a bit while the index is being created, but data consistency should be unaffected in theory

Reply:
Have you seen this trick that I made for the consistency level configuration.. So up until now the local quorum was anyway the default consistency in V1, so effectively nothing changes here. However, you can define your own levels via configuration for schema changes, reads and writes.. I think this is quite a good feature. What do you think?